### PR TITLE
Fix search for users

### PR DESCRIPTION
**Pull request by veolys-nanne**

### CODE REVIEW

File: Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+     * @param string $name
+     * @param string $value
+     * @param string $documentName
+     *
+     * @return array
+     */
+    public function generateFilter($name, $value, $documentName)
+    {
+        $columnsTree = explode('.', $name);
+        if(count($columnsTree) == 2) {
```
Comment (alavieille) [inline Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php]: psr2

Comment (nicolasThal) [inline Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php]: yoda notation

File: Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            list($property, $referenceProperty) = $columnsTree;
+
+            $metadata = $this->documentManager->getClassMetadata($documentName);
+            $targetDocument = $metadata->getFieldMapping($property)['targetDocument'];
+
+            $mapping = $this->searchMappingReader->extractMapping($targetDocument);
+
+            $repository = $this->documentManager->getRepository($targetDocument);
+            $repository->setAggregationQueryBuilder($this->aggregationQueryBuilder);
```
Comment (nicolasThal) [inline Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php]: maybe you should check if the repository is an instance of `AbstractAggregateRepository`

File: Pagination/Tests/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategyTest.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    /**
+     * Test get name
+     */
+    public function testGetName()
+    {
+        $this->assertEquals('reference_filter', $this->strategy->getName());
+    }
+}
+
+
```
Comment (alavieille) [inline Pagination/Tests/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategyTest.php]: remove this empty line

**Comment by alavieille** [comment]
:+1:

**Pull request merged**